test/suites/test_ssl.py

```diff
@@ -0,0 +1,348 @@
+            SslTestCase(
+                name="key_crt_ca_ciphers_server_and_key_crt_ca_client",
+                ok=True,
+                server_key_file=self.key_file,
+                server_cert_file=self.cert_file,
+                server_ca_file=self.ca_file,
+                server_ciphers="ECDHE-RSA-AES256-GCM-SHA384",
+                client_key_file=self.key_file,
+                client_cert_file=self.cert_file,
+                client_ca_file=self.ca_file),
+            SslTestCase(
+                name="key_crt_ca_ciphers_server_and_client",
+                ok=True,
+                server_key_file=self.key_file,
+                server_cert_file=self.cert_file,
+                server_ca_file=self.ca_file,
+                server_ciphers="ECDHE-RSA-AES256-GCM-SHA384",
+                client_key_file=self.key_file,
+                client_cert_file=self.cert_file,
+                client_ca_file=self.ca_file,
+                client_ciphers="ECDHE-RSA-AES256-GCM-SHA384"),
+            SslTestCase(
+                name="non_equal_ciphers",
+                ok=False,
+                server_key_file=self.key_file,
+                server_cert_file=self.cert_file,
+                server_ca_file=self.ca_file,
+                server_ciphers="ECDHE-RSA-AES256-GCM-SHA384",
+                client_key_file=self.key_file,
+                client_cert_file=self.cert_file,
+                client_ca_file=self.ca_file,
+                client_ciphers="TLS_AES_128_GCM_SHA256"),
+        ]
+        for t in testcases:
+            with self.subTest(msg=t.name):
+                srv = None
+                con = None
+                try:
+                    srv = TarantoolServer(
+                        transport = "ssl",
+                        ssl_key_file = t.server_key_file,
+                        ssl_cert_file = t.server_cert_file,
+                        ssl_ca_file = t.server_ca_file,
+                        ssl_ciphers = t.server_ciphers)
+                    srv.script = 'test/suites/box.lua'
+                    srv.start()
+                    srv.admin("box.schema.create_space('space_1')")
+                    srv.admin("""
+                    box.space['space_1']:create_index('primary', {
+                        type = 'tree',
+                        parts = {1, 'num'},
+                        unique = true})
+                    """.replace('\n', ' '))
+                    srv.admin("""
+                    box.schema.user.create('test', { password = 'test' })
+                    """.replace('\n', ' '))
+                    srv.admin("""
+                    box.schema.user.grant('test', 'execute,read,write', 'universe')
+                    """.replace('\n', ' '))
+                    con = tarantool.Connection(
+                        srv.host, srv.args['primary'],
+                        user = "test",
+                        password = "test",
+                        transport = "ssl",
```

Comment:
Shouldn't we have a test case, where a server is configured to accept unencrypted connections, but the client is configured to use TLS. The client should refuse the connection in the case.

Reply:
Sure, added.